### PR TITLE
parse reusable workflow and add enhancers layer

### PR DESCRIPTION
**Pull request by tamirkiviti13**
## Description

This PR adds support for parsing GitHub workflows calls.

## Related issues
- Close #XXX

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/argonsecurity/pipeline-parser/blob/main/CONTRIBUTING.md) to this repository.
- [x] I've followed the [conventions](https://github.com/argonsecurity/pipeline-parser/blob/main/CONTRIBUTING.md#pull-requests) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [readme](https://github.com/argonsecurity/pipeline-parser/blob/main/README.md) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,6 +15,3 @@ jobs:
 
       - name: Execute Tests
         run: make test-coverage
-
-      - name: Upload Coverage
```
Comment (ShiraCohen33) [inline .github/workflows/test.yml:19]: why did you delete it?

File: pkg/enhancers/github/github.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	var errs error
+	importedPipelines, err := getReusableWorkflows(data, credentials)
+	if err != nil {
+		errs = errors.Wrap(errs, err.Error())
+	}
+
+	return importedPipelines, errs
+}
+
+func (g *GitHubEnhancer) Enhance(data *models.Pipeline, importedPipelines []*enhancers.ImportedPipeline, credentials *models.Credentials) (*models.Pipeline, error) {
```
Comment (ShiraCohen33) [inline pkg/enhancers/github/github.go:21]: `credentials *models.Credentials` it seems you don't use this param in this function

File: test/fixtures/github/workflow-call.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+name: workflow-call
+
+jobs:
+  call-local-workflow:
+    uses: ./../fixtures/github/dependant-jobs.yaml
```
Comment (ShiraCohen33) [inline test/fixtures/github/workflow-call.yaml:5]: can you check workflow call on remote?

File: pkg/parsers/github/jobs.go [hunk 4]
```diff
@@ -104,3 +171,49 @@ func parseDependencies(needs *githubModels.Needs) []*models.JobDependency {
...
+
+	if strings.HasPrefix(uses, "./") {
+		path = uses
+		versionType = models.None
+		sourceType = models.SourceTypeLocal
+		return
+	}
+
+	result := githubWorkflowCallRegex.FindStringSubmatch(uses)
+	if len(result) == 0 {
```
Comment (orizerah) [inline pkg/parsers/github/jobs.go:204]:
```suggestion
	if len(result) < 5 {
```

File: pkg/utils/file.go
Comment (orizerah) [inline pkg/utils/file.go:1]: Destroy

**Review by orizerah — COMMENTED** [review]
nice

File: .github/workflows/test.yml (hunk 1 above)
Comment (tamirkiviti13) [reply .github/workflows/test.yml:19]: Originally it was supposed to upload our code coverage to codecov and add a badge to the readme.
It never worked, because of some configurations that we needed to do in the organization level that we never did, so I removed it.

File: test/fixtures/github/workflow-call.yaml (hunk 3 above)
Comment (tamirkiviti13) [reply test/fixtures/github/workflow-call.yaml:5]: I am not sure I understand the question - in line 10 of this file we have a check for remote workflow call

**Review by orizerah — APPROVED** [review]

**Review by orizerah — APPROVED** [review]
